types/index.d.ts

```diff
@@ -0,0 +1,369 @@
+
+    /** Updates and immediately persists a single attr on this model. */
+    update<K extends keyof Data>(key: K, value: Data[K]): void;
+
+    /** Removes this model from the Mirage database. */
+    destroy(): void;
+
+    /** Reloads this model's data from the Mirage database. */
+    reload(): void;
+  };
+}
+
+declare module "miragejs/server" {
+  import { Request, Response } from "miragejs";
+  import { ModelInstance } from "miragejs/-types";
+  import Db from "miragejs/db";
+  import Schema from "miragejs/orm/schema";
+
+  type MaybePromise<T> = T | PromiseLike<T>;
+
+  /** A callback that will be invoked when a given Mirage route is hit. */
+  export type RouteHandler<Registry> = (
+    schema: Schema<Registry>,
+    request: Request
+  ) => MaybePromise<ModelInstance | Response | object>;
+
+  export interface HandlerOptions {
+    /** A number of ms to artifically delay responses to this route. */
+    timing?: number;
+  }
+
+  export class Server<Registry = Record<string, ModelInstance>> {
+    /** The underlying in-memory database instance for this server. */
+    readonly db: Db;
+
+    /** An interface to the Mirage ORM that allows for querying and creating records. */
+    readonly schema: Schema<Registry>;
+
+    /** Creates a model of the given type. */
+    readonly create: Schema<Registry>["create"];
+
+    /** Whether or not Mirage should log all requests/response cycles. */
+    logging: boolean;
+
+    /** A default number of ms to artifically delay responses for all routes. */
+    timing: number;
+
+    /** A default prefix applied to all subsequent route definitions. */
+    namespace: string;
+
+    /** Creates multiple models of the given type. */
+    createList<
+      K extends keyof Registry,
+      Init extends Registry[K],
+      Data extends Partial<Init>
+    >(modelName: K, count: number, data?: Data): Array<Init & Data>;
+
+    /** Handle a GET request to the given path. */
+    get(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a POST request to the given path. */
+    post(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a PUT request to the given path. */
+    put(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a PATCH request to the given path. */
+    patch(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Handle a DELETE request to the given path. */
+    del(
+      path: string,
+      handler?: RouteHandler<Registry>,
+      options?: HandlerOptions
+    ): void;
+
+    /** Pass through one or more URLs to make real requests. */
+    passthrough(...urls: string[]): void;
+
+    /** Load all available fixture data matching the given name(s). */
+    loadFixtures(...names: string[]): void;
```

Comment:
Even though this method exists, there's also the new `seeds` method that does the same with (imho) more explicit naming, better to keep both for now.

The method `routes` was also added in the meantime

Reply:
`seeds` and `routes` are not really listed in the documentation. https://miragejs.com/api/classes/server#load-fixtures

It is mentioned in the comment section of `loadFixtures`, but not really clear how could you use it.

As I see in the source code, `routes` and `seeds` can be a property on the `config` object.

https://github.com/miragejs/miragejs/blob/master/lib/server.js#L222-L236

Reply:
Anyway, methods added without description.